wpsc-includes/country-api.php

```diff
@@ -0,0 +1,193 @@
+ */
+function wpsc_get_region_object( $region_id ) {
+	return WPSC_Countries::get_region( false, $region_id );
+}
+
+/**
+ * Chaneg the cisibility of all countries site wide
+ *
+ * @param int the WP eCommerce specific integer region id
+ *
+ * @since 4.1
+ *
+ * @return WPSC_Country|boolean the country, or false on failure
+ */
+function wpsc_set_all_countries_visibility( $visibility = true ) {
+	WPSC_Countries::set_visibility( $visibility );
+}
+
+/**
+ * Get countries
+ *
+ * @param bool $include_invisible include the invisible countries in the result
+ * @param bool $sort_by_name  sort the results by the country name
+ *
+ * @since 4.1
+ *
+ * @return WPSC_Country[]
+ *
+ */
+function wpsc_get_country_objects( $include_invisible = false, $sort_by_name = true ) {
+	return WPSC_Countries::get_countries( $include_invisible, $sort_by_name );
+}
+
+/**
+ * Get all countries
+ *
+ * @param bool $include_invisible include the invisible countries in the result
+ * @param bool $sort_by_name  sort the results by the country name
+ *
+ * @since 4.1
+ *
+ * @return WPSC_Country[]
+ *
+ */
+function wpsc_get_all_countries( $sort_by_name = true ) {
+	return WPSC_Countries::get_countries( true, $sort_by_name );
+}
+
+/**
+ * Get visible countries
+ *
+ * @param bool $include_invisible include the invisible countries in the result
+ * @param bool $sort_by_name  sort the results by the country name
+ *
+ * @since 4.1
+ *
+ * @return WPSC_Country[]
+ *
+ */
+function wpsc_get_visible_countries( $sort_by_name = true ) {
+	return WPSC_Countries::get_countries( false, $sort_by_name );
+}
+
+/**
+ * Get all known regions
+ *
+ * @since 4.1
+ *
+ * @return WPSC_Region[]
+ *
+ */
+function wpsc_get_all_regions() {
+	return WPSC_Countries::get_all_regions();
+}
+
+/**
+ * Get the country object correcposnding to the currently configured store currency type, might be the default
+ * currency type if store admin has not configured currency type
+ *
+ * @since 4.1
+ *
+ * @return WPSC_Country|boolean the country, or false on failure
+ */
+function wpsc_get_currency_type_country_object() {
+
+	$wpsc_country_of_currency_type = false;
+
+	$currency_type_country_id = intval( get_option( 'currency_type', - 1 ) );
+
+	// get the country for the current currency type, make sure it is valid, if not
+	// we will move on as if a currency type was not set
+	if ( - 1 != $currency_type_country_id ) {
+		$wpsc_country_of_currency_type = wpsc_get_country_object( $currency_type_country_id );
+		if ( ! $wpsc_country_of_currency_type ) {
+			$currency_type_country_id = - 1;
+		}
+	}
+
+	if ( - 1 == $currency_type_country_id ) {
+		$wpsc_country_of_currency_type = wpsc_get_country_object( 'US' );
+		if ( $wpsc_country_of_currency_type ) {
+			$currency_type_country_id = $wpsc_country_of_currency_type->get_id();
+		}
+	}
+
+	return $wpsc_country_of_currency_type;
+}
+
+
+/**
+ * Get the country id correcposnding to the currently configured store currency type, might be the default
```

Comment:
Docs need a grammar/spelling pass.

Also, this function name is sort of obtuse.  I get what it's doing, but it just feels a bit much.  Not sure what I'd change just yet, but i feel like a lot of this PR suffers from [number two](http://martinfowler.com/bliki/TwoHardThings.html)